picard/ui/options/maintenance.py

```diff
@@ -154,6 +161,116 @@ def open_config_dir(self):
         config_dir = path.split(config.fileName())[0]
         QtGui.QDesktopServices.openUrl(QtCore.QUrl.fromLocalFile(config_dir))
 
+    def _get_dialog_filetypes(self, _ext='.ini'):
+        return ";;".join((
+            _("Configuration files") + " (*{0})".format(_ext,),
+            _("All files") + " (*)",
+        ))
+
+    def _make_backup_filename(self, auto=False):
+        config = get_config()
+        _filename = path.split(config.fileName())[1]
+        _root, _ext = path.splitext(_filename)
+        return "{0}_{1}_Backup_{2}{3}".format(
+            _root,
+            'Auto' if auto else 'User',
+            datetime.datetime.now().strftime("%Y%m%d_%H%M"),
+            _ext,
+        )
+
+    def _backup_error(self, dialog_title=None):
+        if not dialog_title:
+            dialog_title = _("Backup Configuration File")
+        dialog = QtWidgets.QMessageBox(
+            QtWidgets.QMessageBox.Icon.Critical,
+            dialog_title,
+            _("There was a problem backing up the configuration file. Please see the logs for more details."),
+            QtWidgets.QMessageBox.StandardButton.Ok,
+            self
+        )
+        dialog.exec_()
+
+    def save_backup(self):
+        config = get_config()
+        directory = path.normpath(QtCore.QStandardPaths.writableLocation(QtCore.QStandardPaths.StandardLocation.DocumentsLocation))
+        filename = self._make_backup_filename()
+        ext = path.splitext(filename)[1]
+        default_path = path.normpath(path.join(directory, filename))
+
+        dialog_title = _("Backup Configuration File")
+        dialog_file_types = self._get_dialog_filetypes(ext)
+        options = QtWidgets.QFileDialog.Options()
+        filename, file_type = QtWidgets.QFileDialog.getSaveFileName(self, dialog_title, default_path, dialog_file_types, options=options)
+        if not filename:
+            return
+        # Fix issue where Qt may set the extension twice
+        (name, ext) = path.splitext(filename)
+        if ext and str(name).endswith('.' + ext):
+            filename = name
+
+        if config.save_user_backup(filename):
+            dialog = QtWidgets.QMessageBox(
+                QtWidgets.QMessageBox.Icon.Information,
+                dialog_title,
+                _("Configuration successfully backed up to %s") % filename,
+                QtWidgets.QMessageBox.StandardButton.Ok,
+                self
+            )
+            dialog.exec_()
+        else:
+            self._backup_error(dialog_title)
+
+    def load_backup(self):
+        dialog_title = _("Load Backup Configuration File")
+        dialog = QtWidgets.QMessageBox(
+            QtWidgets.QMessageBox.Icon.Warning,
+            dialog_title,
+            _("Loading a backup configuration file will replace the current configuration settings. "
+            "A backup copy of the current file will be saved automatically.\n\nDo you want to continue?"),
```

Comment:
@zas Is this an acceptable way to combine parts of a long string for translation (so as not to have an extremely long line)?

Reply:
Yes, they'll appear as one string in .po files